WcaOnRails/app/webpacker/components/CompetitionOverview/CompetitionTable.js

```diff
@@ -0,0 +1,190 @@
+        position="top center"
+        size="tiny"
+      />
+    );
+  }
+  if (comp.registration_status === 'full') {
+    return (
+      <Popup
+        trigger={<Icon className="user clock orange" />}
+        content={I18n.t('competitions.index.tooltips.registration.full')}
+        position="top center"
+        size="tiny"
+      />
+    );
+  }
+
+  return (
+    <Popup
+      trigger={<Icon className="user plus green" />}
+      content={I18n.t('competitions.index.tooltips.registration.open')}
+      position="top center"
+      size="tiny"
+    />
+  );
+}
+
+function DateIcon({ comp, showRegistrationStatus, sortByAnnouncement }) {
+  let tooltipInfo = '';
+  let iconClass = '';
+
+  if (comp.isProbablyOver) {
+    if (comp.resultsPosted) {
+      tooltipInfo = I18n.t('competitions.index.tooltips.hourglass.posted');
+      iconClass = 'check circle result-posted-indicator';
+    } else {
+      tooltipInfo = I18n.t('competitions.index.tooltips.hourglass.ended', { days: I18n.t('common.days', { count: calculateDayDifference(comp.start_date, comp.end_date, 'past') }) });
+      iconClass = 'hourglass end';
+    }
+  } else if (comp.inProgress) {
+    tooltipInfo = I18n.t('competitions.index.tooltips.hourglass.in_progress');
+    iconClass = 'hourglass half';
+  } else if (sortByAnnouncement) {
+    tooltipInfo = I18n.t('competitions.index.tooltips.hourglass.announced_on', { announcement_date: comp.announcedDate });
+    iconClass = 'hourglass start';
+  } else if (showRegistrationStatus) {
+    return <RegistrationStatus comp={comp} />;
+  } else {
+    tooltipInfo = I18n.t('competitions.index.tooltips.hourglass.starts_in', { days: I18n.t('common.days', { count: calculateDayDifference(comp.start_date, comp.end_date, 'future') }) });
+    iconClass = 'hourglass start';
+  }
+
+  return (
+    <Popup
+      trigger={<Icon className={iconClass} />}
+      content={tooltipInfo}
+      position="top center"
+      size="tiny"
+    />
+  );
+}
+
+// Currently, the venue attribute of a competition object can be written as markdown,
+// and using third party libraries like react-markdown to parse it requires to much work
+function VenueMarkdown({ venueText }) {
+  const openBracketIndex = venueText.indexOf('[');
+  const closeBracketIndex = venueText.indexOf(']', openBracketIndex);
+  const openParenIndex = venueText.indexOf('(', closeBracketIndex);
+  const closeParenIndex = venueText.indexOf(')', openParenIndex);
+
+  if (openBracketIndex === -1 || closeBracketIndex === -1
+    || openParenIndex === -1 || closeParenIndex === -1) {
+    return <p>{venueText}</p>;
+  }
+
+  return (
+    <a href={venueText.slice(openParenIndex + 1, closeParenIndex)} target="_blank" rel="noreferrer">
+      <p>{venueText.slice(openBracketIndex + 1, closeBracketIndex)}</p>
+    </a>
+  );
+}
+
+function LoadedDisplay({ numCompetitions }) {
```

Comment:
```suggestion
function EndOfCompListMessage({ numCompetitions }) {
```
or something would be better - it's usually best to describe what the component is doing rather than when/where it's used.

When I first saw this being called in `CompetitionTable` I thought it must be displaying data (because of the name, and since it's only called when `loaded` is true), which was confusing.

Reply:
helpful suggestions, didn't see them at first so I can no longer commit them here but I added them now.